### ISSUE TITLE
UWP - after some days being installed, app crashes at startup as JobManager.SheduleNative throws System.Exception "Quota Exceeded"

### ISSUE THREAD

**Issue by gentledepp** [shinyorg/shiny#511]
## Please check all of the platforms you are having the issue on (if platform is not listed, it is not supported)

 - [ ] iOS
 - [ ] Android
 - [x] UWP

 PLATFORMS THAT YOU CANNOT FILE ISSUES ON RIGHT NOW: Tizen, watchOS, tvOS, macOS

### beta components are not fully done - issues MUST include full reproducible sample and stacktrace or they will be closed without explanation!!!

 - [x] Core
 - [ ] Locations
 - [ ] HTTP Transfers
 - [ ] Notifications
 - [ ] Push
 - [ ] Push - Azure Notification Hub
 - [ ] Push - Firebase (beta)
 - [ ] Sensors
 - [ ] AppCenter Logging Integration
 - [ ] Location Sync (beta)
 - [ ] Photo Sync (alpha)
 - [ ] BluetoothLE (beta)
 - [ ] Beacons (beta)
 - [ ] SQLite Integration

## Version of Library
Library version 1.2.0.1755
## Steps to Reproduce
1. Install app that uses notifications and the background job.
2. after some time, it just crashes (PR with repro will come)

## Expected Behavior
App starts and registers BackgroundJob

## Actual Behavior
App crashes with exception:

![image](https://user-images.githubusercontent.com/1175751/107512864-b7113200-6ba7-11eb-98c1-8cda84a937b8.png)

Interestingly, we cannot find any documentation on UWP as to why this error should be throw or what it means.
Our backgroundjob is registered for every 60 minutes - so it should not be called too often either.

The easiest (and for now only) fix is to catch and ignore the exception:

`Shiny.Core\Jobs\Platforms\Uwp\JobManager.cs` 
```diff
        protected override void ScheduleNative(JobInfo jobInfo)
        {
            this.CancelNative(jobInfo);

            var builder = new BackgroundTaskBuilder();
            builder.Name = GetJobTaskName(jobInfo);
            builder.TaskEntryPoint = typeof(ShinyBackgroundTask).FullName;

            if (jobInfo.PeriodicTime != null)
            {
                if (jobInfo.PeriodicTime < TimeSpan.FromMinutes(15))
                    throw new ArgumentException("You cannot schedule periodic jobs faster than 15 minutes");

                var runMins = Convert.ToUInt32(Math.Round(jobInfo.PeriodicTime.Value.TotalMinutes, 0));
                builder.SetTrigger(new TimeTrigger(runMins, false));
            }

            //SystemTriggerType.PowerStateChange
            // TODO: idle, power change, etc
            if (jobInfo.RequiredInternetAccess != InternetAccess.None)
            {
                var type = jobInfo.RequiredInternetAccess == InternetAccess.Any
                    ? SystemConditionType.InternetAvailable
                    : SystemConditionType.FreeNetworkAvailable;

                builder.AddCondition(new SystemCondition(type));
            }

+           try
+           {
                builder.Register();
+           }
+           catch (Exception)
+           {
+				// todo: log exception
+           }
        }
```

**Comment by gentledepp** [comment]
Update: Apparently, the issue appears after some time, when the backgroundTaskHost.exe fails constantly. Windows EventViewer reveals:
![image](https://user-images.githubusercontent.com/1175751/107513579-b927c080-6ba8-11eb-8e2a-03d332a34b07.png)

**This is weird, as it shows the job failing every couple of minutes, but we scheduled it every 60 minutes!**

The following blog article goes into several reasons for this:
https://appuals.com/0xc000027b/

What the heck.... UWP is really a mess 🙄

**Comment by aritchie** [comment]
Do you have the full exception stack trace? 

While I appreciate the PR, this is only masking the issue.  There is something else weird going on.  It sounds like it is creating too many native jobs somehow.  The stack trace should give me a better view of this. 

Shiny does prevent all errors from the job itself, but this is getting to the event viewer which means Shiny is crashing calling into the OS.

**Comment by gentledepp** [comment]
Well.... yes but I'm afraid you won't like it:

>   at Windows.ApplicationModel.Background.BackgroundTaskBuilder.Register()
>   at Shiny.Jobs.JobManager.ScheduleNative(JobInfo jobInfo) in 
>   C:\code_gh\shiny_oq\src\shiny.core\Jobs\Platforms\Uwp\JobManager.cs:line 85

**Comment by aritchie** [comment]
I believe the issue is that the native job continuously gets re-registered.  This whole process has been rewritten in 2.0 previews.  Can you please try preview and see if you experience this.

**Comment by aritchie** [comment]
I haven't been able to repro this in 2.0 so closing this out

**Comment by gentledepp** [comment]
Sure! Hopefully I will find some minutes tomorrow to test what you proposed. I will ping back and let you know if that is indeed the solution.

**Comment by gentledepp** [comment]
Well... I just tried to upgrade to the preview packages.
There is a bunch of compile errors that I had to dig through.
Unfortunately, now the application does not start anymore.
Seems I did something wrong, but as for my lack of time and the fact, that I could not find any documentation or even samples for how to use shiny 2.x ++ I will leave it as is for now.

Once there is a documented beta (or one with samples), I will revisit this and try again.
